core/src/main/java/com/graphhopper/routing/AlternativeRouteCH.java

```diff
@@ -128,15 +132,15 @@ Path extract() {
                 return true;
             }
 
-            private double calculateShare(final Path path) {
-                double sharedDistance = 0.0;
-                List<EdgeIteratorState> edges = path.calcEdges();
-                for (EdgeIteratorState edge : edges) {
-                    if (nodesInCurrentAlternativeSetContains(edge.getBaseNode()) && nodesInCurrentAlternativeSetContains(edge.getAdjNode())) {
-                        sharedDistance += edge.getDistance();
+            private double calculateShare(final List<SPTEntry> sptEntries) {
+                double sharedWeight = 0.0;
+                for (int i = 0; i < sptEntries.size(); i++) {
```

Comment:
Maybe for-each loop here or is it really faster that way ?

Reply:
After some micro benchmarks a few years ago I avoid this for-each loop. E.g. in EncodingManager.reverseFlag there was a change that improved routing speed when replacing the for-each loop once.